### ISSUE TITLE
Launching Rich Presence when TF2 is launched on Steam

### ISSUE THREAD

**Issue by Keanu73** [Kataiser/tf2-rich-presence#45]
Would it be possible to replace the main TF2 executable with a shell executable that wraps around the main game providing rich presence? I might be wrong here so forgive me

**Comment by Kataiser** [comment]
That's a good suggestion, and it may also be possible. I'll look into it.

**Comment by Keanu73** [comment]
Although a downside to this is that if the exe is updated when a new TF2 update hits, you'll have to do the same process again of copying the main TF2 executable somewhere else, then pasting the wrapper executable.

**Comment by Keanu73** [comment]
Any progress on this?

**Comment by Kataiser** [comment]
I began to implement it a while ago, but it occurred to me that "modifying" hl2.exe could trigger VAC, so I stopped development.

**Comment by Keanu73** [comment]
Yes, I know that modifying hl2.exe can trigger VAC, which is why I suggested instead of hooking it with some arbitrary DLL, just replace it with another executable.

**Comment by barrulik** [comment]
i made that bat code to make it happend.

how to do it in steps

**step 1:**
add that bat file wherever you want, just make sure u know the file path, the file path should look like C:\home\sally\run.bat.

**step 2:**
after that, click the properties button on the team fortress 2 on the cd "%appdata%\..\Local\Programs\steam" path, 
![image](https://user-images.githubusercontent.com/62114976/132916319-af16413e-4715-4211-8d53-32b51353e21c.png)
then change the URL to your bat file path and click apply and ok
![image](https://user-images.githubusercontent.com/62114976/132916500-a79e3b24-bf1b-4b62-ba9b-7fe17c94bbb3.png)

now if u run tf2, it will open rich presence no matter what

`
cd "%appdata%\..\Local\Programs\TF2 Rich Presence"
call "%appdata%\..\Local\Programs\TF2 Rich Presence\TF2 Rich Presence.bat"
timeout 2
call explorer steam://rungameid/440
exit
`

**Comment by barrulik** [comment]
btw @Keanu73 
its my first time trying that program out

**Comment by Kataiser** [comment]
Oh cool! This still won't affect launching the game via Steam, but I'll consider adding a feature to let you either create a shortcut like this automatically, or modify your start menu shortcut like you've done. Probably should also add a launch parameter to automatically open TF2 with -condebug.

**Comment by barrulik** [comment]
I guess i just need to add that after the "call" that calls ur program

**Comment by Kataiser** [comment]
No I'm saying I need to add that parameter to my program

**Comment by barrulik** [comment]
Oh, i will try making that, keep in mind im not so good in python...

**Comment by Kataiser** [comment]
I appreciate the initiative, but... I literally just said I'd do it...

**Comment by barrulik** [comment]
I missread it oops, maybe i shouldnt look in github while class in phisical school

**Comment by Kataiser** [comment]
Probably lol
